cityofaustin/atd-moped
# Only query for columns shown in the project list view

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/14298

This PR updates the code to only fetch the data associated with columns shown in the project list view.

## Testing
**URL to test:** 

https://deploy-preview-1209--atd-moped-main.netlify.app/

**Steps to test:**
1. Go to the https://deploy-preview-1209--atd-moped-main.netlify.app/moped/projects
2. Open dev tools because the easiest way to visualize this change is to inspect the GraphQL queries in the network requests
3. Refresh the page and check the columns in the request and check that the columns fetched match up with the columns checked in the column selector UI
   - The config in `PROJECT_LIST_VIEW_QUERY_CONFIG.columns` and the `useColumns` hook can be helpful to match column names to the column labels
   - Also, expect to see `current_phase_key` in each request since it is needed to display the status badge
4. Add/remove columns and inspect requests to see more/less fields fetched
5. Test the `useCurrentData` hook by reloading the page
   - You should see a spinner on first load as the first page is fetching
   - When changing filters, sorting, paginating, or updating columns, you should see no spinner or UI changes other than rows of data updating
6. Test the hidden column preferences by watching the value of `mopedColumnConfig` in your browser's local storage to see columns switch between true and false when toggling them in the table

---
#### Ship list
